listenbrainz/webserver/static/js/src/stats/UserStats.tsx

```diff
@@ -0,0 +1,130 @@
+import * as ReactDOM from "react-dom";
+import * as React from "react";
+
+import ErrorBoundary from "../ErrorBoundary";
+import Pill from "../components/Pill";
+import UserListeningActivity from "./UserListeningActivity";
+
+export type UserStatsProps = {
+  user: ListenBrainzUser;
+  apiUrl: string;
+};
+
+export type UserStatsState = {
+  range: UserStatsAPIRange;
+};
+
+class UserStats extends React.Component<UserStatsProps, UserStatsState> {
+  constructor(props: UserStatsProps) {
+    super(props);
+
+    this.state = {
+      range: "week",
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("popstate", this.syncStateWithURL);
+
+    const range = this.getURLParams();
+    window.history.replaceState(null, "", `?range=${range}`);
+    this.syncStateWithURL();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("popstate", this.syncStateWithURL);
+  }
+
+  changeRange = (newRange: UserStatsAPIRange): void => {
+    this.setURLParams(newRange);
+    this.syncStateWithURL();
+  };
+
+  syncStateWithURL = async (): Promise<void> => {
+    const range = this.getURLParams();
+    this.setState({ range });
+  };
+
+  getURLParams = (): UserStatsAPIRange => {
+    const url = new URL(window.location.href);
+
+    let range: UserStatsAPIRange = "week";
+    if (url.searchParams.get("range")) {
+      range = url.searchParams.get("range") as UserStatsAPIRange;
+    }
+
+    return range;
+  };
+
+  setURLParams = (range: UserStatsAPIRange): void => {
+    window.history.pushState(null, "", `?range=${range}`);
+  };
+
+  render() {
+    const { range } = this.state;
+    const { apiUrl, user } = this.props;
+
+    return (
+      <div>
+        <div className="row mt-15">
+          <div className="col-xs-12">
+            <Pill
```

Comment:
could we make this  selector an independent component and then reuse it on the entity page somehow? Basically a component that takes a list of options and callbacks and generates a selector? 

Not sure if this would be overengineering it too much.

Reply:
I think it's fine as it is, we would have to pass an `onClick` handler for each option as it is, so it's not gonna reduce the amount of code significantly but would make it hard to customise the pill further.